### PR TITLE
Miscellaneous changes regarding SDP support

### PR DESCRIPTION
**Pull request by etienne-lms**
Add buidl test with SDP enabled in travis.

Define a default SDP memory test area for STM platforms. This eases testing SDP on b2260.

A internal routine labeling cleanup kindly reported by jerome.

### CODE REVIEW

File: core/arch/arm/plat-stm/platform_config.h [hunk 1]
```diff
@@ -282,13 +288,30 @@
...
 #if defined(CFG_WITH_PAGER)
 
 #define TZSRAM_BASE		CFG_CORE_TZSRAM_EMUL_START
 #define TZSRAM_SIZE		CFG_CORE_TZSRAM_EMUL_SIZE
 
 #define TZDRAM_BASE		CFG_DDR_TEETZ_RESERVED_START
-#define TZDRAM_SIZE		(CFG_DDR_TEETZ_RESERVED_SIZE - CFG_SHMEM_SIZE)
+#define TZDRAM_SIZE		(CFG_DDR_TEETZ_RESERVED_SIZE - \
+				CFG_SHMEM_SIZE - \
+				CFG_TEE_SDP_TEST_MEM_SIZE)
```
Comment (jforissier) [inline core/arch/arm/plat-stm/platform_config.h]: `CFG_TEE_SDP_TEST_MEM_SIZE` is undefined here when both `CFG_SECURE_DATA_PATH` and `CFG_TEE_SDP_MEM_BASE` are defined, no?

File: core/arch/arm/plat-stm/platform_config.h [hunk 2]
```diff
@@ -299,7 +322,9 @@
 #else  /* CFG_WITH_PAGER */
 
 #define TZDRAM_BASE		CFG_DDR_TEETZ_RESERVED_START
-#define TZDRAM_SIZE		(CFG_DDR_TEETZ_RESERVED_SIZE - CFG_SHMEM_SIZE)
+#define TZDRAM_SIZE		(CFG_DDR_TEETZ_RESERVED_SIZE - \
+				CFG_SHMEM_SIZE - \
+				CFG_TEE_SDP_TEST_MEM_SIZE)
```
Comment (jforissier) [inline core/arch/arm/plat-stm/platform_config.h]: Same here

File: core/arch/arm/plat-stm/platform_config.h (hunk 1 above)
Comment (etienne-lms) [inline core/arch/arm/plat-stm/platform_config.h]: you are right. should be fixed now.
thanks.
Same below.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
comments addressed, squashed, tags applied.

**Pull request merged**